[W10-4] Know-It-All

afterdusk opened the pull request:
@dlqs @KerrynEer @mmdlow @yichong96

kylase-learning commented:
## Feedback (Week 7; v1.1)
 
### Github Project Page
- [x] Update Project description
- [x] Update Link to Project Page
 
### README
- [x] UI Mockup updated in project website
- [ ] Make the project website look like a product home page (aka remove the top nav bar)
- [x] AB4 ackhowledgement
- [x] Travis badge updated to own fork's
- [x] Coveralls badge updated to own fork's
 
### [Project Page](https://cs2103-ay1819s2-w10-4.github.io/main/)
- [x] Update to remove AB4 reference
- [x] Travis badge updated to own fork's
- [x] Coveralls badge updated to own fork's

### About Us Page
- [x] Updated with photos
- [x] Roles and components assigned

UG/DG feedback will be provided to you in the documents.

kylase-learning commented on README.adoc:
```diff
@@ -15,26 +13,64 @@ ifndef::env-github[]
+*_Know-It-All_* is proud to present the following _awards_:
+
+ifdef::env-github[]
+image:award1.png[width="200"] {nbsp}{nbsp}{nbsp}{nbsp}{nbsp}{nbsp}{nbsp}{nbsp}{nbsp}{nbsp}image:award2.png[width="200"]
```
Erm.... is this fake news? 😅

afterdusk committed:
Update CSV test file names

KerrynEer committed:
Merge pull request #26 from cs2103-ay1819s2-w10-4/master

Update UG, DG and PPP and check for valid folder names

yichong96 committed:
rename PPP ascii doc

yichong96 committed:
update PPP intro

afterdusk committed:
Add data directory warning to QuickStart in UG

KerrynEer committed:
Replace previous saveLocation in status bar to currentStatus

KerrynEer committed:
Replace saveLocation with current status in comment

afterdusk committed:
Update ModelManagerTest and add AddFolderCommandTest

KerrynEer committed:
Add test command logic junit and integration test for an empty folder

KerrynEer committed:
Add next command test for the case when there are no more cards in the folder.

KerrynEer committed:
Fix set current tested Card test method name typo

KerrynEer committed:
Correct mistake in set current tested card test method in ModelManager

dlqs committed:
add test for sort command

afterdusk committed:
Update StorageManagerTest

dlqs committed:
add comparator test

dlqs committed:
add test for end report

dlqs committed:
fix checkstyle

dlqs committed:
add system test for report

afterdusk committed:
Add EditFolderTest

Also fixed bug with rename() method in CardFolder and performed minor updates to other tests

afterdusk committed:
Fix wrong invalid index message for EditFolderCommand

afterdusk committed:
Add DeleteFolderCommandTest

afterdusk committed:
Update CardFolderTest

afterdusk committed:
Fix resetData() test bug in CardFolderTest

KerrynEer committed:
Refactor testCardFolder() to startTestSession() in model

KerrynEer committed:
Fix typo in comment for sort test in model

KerrynEer committed:
Add getters in Model Manager for testing
- get current tested card folder
- get num answered correctly

KerrynEer committed:
Add setup methods for test session related tests

KerrynEer committed:
Add tests for test session ModelManager methods and use assertNull/Non-Null

KerrynEer committed:
Revert commit: add test command logic junit and integration test for an empty folder
- this is tested in modelManager test

afterdusk committed:
Refactor tests and remove redundant ModelManager constructor

afterdusk committed:
Fix bug in StorageManager and update tests accordingly

afterdusk committed:
Merge pull request #177 from afterdusk/master

Perform bug fixes and add tests

KerrynEer committed:
Fix checkstyle

mmdlow committed:
Merge pull request #178 from mmdlow/parser

Restrict number of arguments for add and edit card commands

KerrynEer committed:
Fix merge conflicts and Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/test/java/seedu/knowitall/logic/commands/AnswerCommandTest.java
#	src/test/java/seedu/knowitall/logic/commands/NextCommandTest.java
#	src/test/java/seedu/knowitall/model/ModelManagerTest.java

KerrynEer committed:
Fix error due to duplicate card folder names for next command test

KerrynEer committed:
Remove old comment

KerrynEer committed:
Refactor some contents in browser panel to flashcard context

afterdusk committed:
Update User Guide Quick Start

afterdusk committed:
Merge remote-tracking branch 'upstream/master'

afterdusk committed:
Further additions to UG Quick Start

KerrynEer committed:
Add images and ppt for user guide

afterdusk committed:
Update UI.png

mmdlow committed:
Add newline characters to card added message

KerrynEer committed:
Update User guide

KerrynEer committed:
Update test session sequence diagram and ui class diagram for DG

KerrynEer committed:
Update DG

KerrynEer committed:
Update PPP

yichong96 committed:
update documentation

yichong96 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-w10-4/main into csvOptionHeader

afterdusk committed:
Update PPP

And minor update to legend in UG

afterdusk committed:
Fix typos and rephrase some sentences

afterdusk committed:
Update Architecture Diagram

KerrynEer committed:
Add manual testing for test session features in DG

KerrynEer committed:
Remove se-edu bar from website

afterdusk committed:
Update Developer Guide

KerrynEer committed:
Update sequence diagrams for execute() to include arguments

KerrynEer committed:
Fix typos in DG and ppp

KerrynEer committed:
Remove space in manual testing instructions in DG

yichong96 committed:
added UI to PPP and update import export code

KerrynEer committed:
Fix typo in PPP

yichong96 committed:
update blood csv test

KerrynEer committed:
Rename Uimockups file to images for user guide

dlqs committed:
add test system test

dlqs committed:
update comment

yichong96 committed:
change design considerations to table format

dlqs committed:
fix comment

KerrynEer committed:
Merge pull request #179 from KerrynEer/master

Add test session related tests and update docs

dlqs committed:
Merge branch 'master' into minTestScore

afterdusk committed:
Fix mistakes in DG and PPP

yichong96 committed:
add preface to design considerations

dlqs committed:
fix naming

afterdusk committed:
Fix merge conflict

dlqs committed:
Merge pull request #167 from dlqs/minTestScore

Min test score

afterdusk committed:
Add manual testing for folders

afterdusk committed:
Merge remote-tracking branch 'upstream/master'

afterdusk committed:
Amend sequence diagrams

afterdusk committed:
Fix checkstyle issues

yichong96 committed:
changedd IMPORTANT notation to NOTE

yichong96 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-w10-4/main into csvOptionHeader

afterdusk committed:
Replace 'the user' with 'you' in UG

afterdusk committed:
Update addfolder command in UG to have UI image

KerrynEer committed:
Merge pull request #180 from afterdusk/master

Update Documentation

dlqs committed:
add capitalization

dlqs committed:
update definition

KerrynEer committed:
Fix typos and add some details to docs

KerrynEer committed:
Fix typos in links in design of DG

KerrynEer committed:
Update phrasing and add command format to PPP

mmdlow committed:
Enable text overflow in card page ui

mmdlow committed:
Add scrollpane to browser panel

dlqs committed:
add new sequence diagrams

dlqs committed:
add explaination for report command

mmdlow committed:
Show browserpanel scrollbar only when needed

dlqs committed:
update my adoc name

mmdlow committed:
Reduce max length of questions, answers, hints and options to 256

dlqs committed:
add link to reposense

dlqs committed:
add formating code snippet

dlqs committed:
add new screenshot for report

KerrynEer committed:
Fix more typos and wordings

KerrynEer committed:
Remove Johndoe PPP

dlqs committed:
fix ordering caption

mmdlow committed:
Update answer command tests for mcq cards

mmdlow committed:
Fix bug with mcq answer test

 - Bug where answering an mcq card with a large number (larger than nnumber of options available) will not throw the command exception telling users to answer with an option number.

KerrynEer committed:
Fix wording in UG Command Format and PPP

mmdlow committed:
Fix bug with import command test

KerrynEer committed:
Fix phrasing in DG

mmdlow committed:
Fix bug with help command system test

mmdlow committed:
Update PPP

mmdlow committed:
Merge branch 'master' into dev

KerrynEer committed:
Fix grammer error in DG

mmdlow committed:
Fix checkstyle

yichong96 committed:
update sequence diagram and PPP

mmdlow committed:
Fix bug in answer command test

yichong96 committed:
fix bug for missing fields csv file

yichong96 committed:
change import command exception message

mmdlow committed:
Update PPP

mmdlow committed:
Center card ui display text alignment

mmdlow committed:
Add images showing the difference between mcq and single answer in UG

mmdlow committed:
Update PPP other contributions

yichong96 committed:
fix parser test

KerrynEer committed:
Fix typo in DG and remove some links in PPP

yichong96 committed:
remove UI of KIA from PPP

KerrynEer committed:
Fix more typos in DG

KerrynEer committed:
Cut down on some information in PPP

yichong96 committed:
reduce documentation from PPP

KerrynEer committed:
Mention in PPP that contributions here are excerpts

mmdlow committed:
Add card field length constraints to user guide

KerrynEer committed:
Add currentTestedCardFolderSize and update sequence diagram

KerrynEer committed:
Shorten ug portion in PPP

KerrynEer committed:
Add tag to PPP

yichong96 committed:
add red text legend

KerrynEer committed:
Bold all instances of Know-It-All in UG

KerrynEer committed:
Fix grammar error

afterdusk committed:
Update jar file link in UG

Co-Authored-By: KerrynEer <39527245+KerrynEer@users.noreply.github.com>

KerrynEer committed:
Merge branch 'master' of https://github.com/KerrynEer/main

KerrynEer committed:
Fix typos in DG

KerrynEer committed:
Merge pull request #183 from KerrynEer/master

Update docs

afterdusk committed:
Update documentation

afterdusk committed:
Merge remote-tracking branch 'upstream/master'

mmdlow committed:
Merge pull request #185 from mmdlow/dev

Fix bugs and update docs

mmdlow committed:
Shrink image sizes in PPP

mmdlow committed:
Merge branch 'master' into dev

yichong96 committed:
added new blood and typical card folder files

yichong96 committed:
update command summary UG for export

yichong96 committed:
fix merge conflict in files

dlqs committed:
add unit test for messages

dlqs committed:
add testing instruction for report

dlqs committed:
add sort manual testing

yichong96 committed:
change import file line endings and added red text for PPP

dlqs committed:
add invalid manual test

dlqs committed:
fix missing images on netlify

dlqs committed:
fix typo

mmdlow committed:
Add card section to manual testing section in developer guide

mmdlow committed:
Restructure manual testing section of developer guide

dlqs committed:
add new v2 feature

mmdlow committed:
Update PPP

mmdlow committed:
Remove images in PPP

mmdlow committed:
Fix bug where check for duplicate cards was not case-insensitive

mmdlow committed:
Shrink size of card comparison images in add card section of UG

mmdlow committed:
Add correct answer ui image to PPP

mmdlow committed:
Fix formatting in PPP

mmdlow committed:
Update user guide to describe duplicate card constraints

dlqs committed:
update ppp

dlqs committed:
fix indent

dlqs committed:
add new pr

dlqs committed:
fix indent

dlqs committed:
fix image

dlqs committed:
Merge pull request #182 from dlqs/updatedoc

Updatedoc

dlqs committed:
Merge pull request #187 from mmdlow/dev

Update docs and fix bugs

afterdusk committed:
Merge remote-tracking branch 'upstream/master'

yichong96 committed:
change line ending unix

KerrynEer committed:
Update manual testing answer command
- now doesnt accept invalid option number

afterdusk committed:
Fix bug with select command message

yichong96 committed:
add appendix f for dev guide

yichong96 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-w10-4/main into csvOptionHeader

yichong96 committed:
update user guide legend

yichong96 committed:
change grey markup legend

dlqs committed:
update doc

KerrynEer committed:
Fix Sequence Diagram For Delete Card image not showing

mmdlow committed:
Set fxid of browserpanel scrollpane

dlqs committed:
change pos

dlqs committed:
typo

afterdusk committed:
Update report display image in UG

dlqs committed:
add newline

afterdusk committed:
Add new line at EOF :) :)

dlqs committed:
change to tip

KerrynEer committed:
Fix typo in DG

yichong96 committed:
change DG design consideration table format

dlqs committed:
change to box

yichong96 committed:
fix checkstyle bugs

mmdlow committed:
Set browserpanel scrollbar to be always present

mmdlow committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-w10-4/main into dev

dlqs committed:
change note

mmdlow committed:
Remove browserpanel scrollbar and enable click and drag for card ui

afterdusk committed:
Update User Guide

yichong96 committed:
fix unwanted symbols

dlqs committed:
uppercase

mmdlow committed:
Merge pull request #191 from mmdlow/dev

Adjust BrowserPanel UI for cards

yichong96 committed:
fix additional bugs

afterdusk committed:
Merge remote-tracking branch 'upstream/master'

KerrynEer committed:
Fix all links in DG

yichong96 committed:
add collated code preface

yichong96 committed:
add back csv file structure in dev guide

afterdusk committed:
Fix UI.png naming error

dlqs committed:
Merge pull request #186 from afterdusk/master

Update documentation

mmdlow committed:
Merge pull request #190 from dlqs/moredoc

update doc

yichong96 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-w10-4/main into csvOptionHeader

KerrynEer committed:
Update NFRs

yichong96 committed:
Merge pull request #181 from yichong96/csvOptionHeader

Csv option header

KerrynEer committed:
Fix UI images naming in README

KerrynEer committed:
Fix UG example format in add command

KerrynEer committed:
Merge remote-tracking branch 'upstream/master'

KerrynEer committed:
Fix typo in DG

yichong96 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-w10-4/main into csvOptionHeader

afterdusk committed:
Make AnswerCommand and RevealCommand not commit to state history

yichong96 committed:
edit documentation

KerrynEer committed:
Update UG to reflect test session undoability

KerrynEer committed:
Revert "Update UG to reflect test session undoability"

This reverts commit e4e29dacf1353811dd388fd756dce5e4a4cfb1ee.

afterdusk committed:
Undo previous commit and make Test Session clear state history

KerrynEer committed:
Update undo command and test operations docs
- now no longer undoable

yichong96 committed:
update dev guide

yichong96 committed:
modified glossary

KerrynEer committed:
Bold all instances of Know-it-all in UG

KerrynEer committed:
Fix typo

afterdusk committed:
Make folder names case insensitive

afterdusk committed:
Merge remote-tracking branch 'upstream/master'

mmdlow committed:
Update glossary references in user guide

afterdusk committed:
Rename a variable in EditFolderCommand

KerrynEer committed:
Fix small typo in DG

mmdlow committed:
Remove errors in command summary of user guide

mmdlow committed:
Fix minor error in dev guide

mmdlow committed:
Update PPP

afterdusk committed:
Fix checkstyle issues

mmdlow committed:
Merge pull request #192 from yichong96/updateDocs

Update docs

KerrynEer committed:
Revert "Bold all instances of Know-it-all in UG"

This reverts commit 8f3dcf2e3d031d3dcc52a23c262b1a8326095487.

mmdlow committed:
Merge branch 'master' into dev

mmdlow committed:
Merge pull request #194 from mmdlow/dev

Update docs

afterdusk committed:
Fix error in resetStates() of VersionedCardFolder

KerrynEer committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/DeveloperGuide.adoc

dlqs committed:
Merge pull request #193 from afterdusk/master

Perform Bug Fixes

dlqs committed:
fix wrong size committed

KerrynEer committed:
Fix typo in link

yichong96 committed:
Merge pull request #189 from KerrynEer/master

Final updates to docs

dlqs committed:
fix div by zeo bug

dlqs committed:
liang's aasserts

dlqs committed:
Merge branch 'master' of github.com:cs2103-ay1819s2-w10-4/main into bugfix

dlqs committed:
yc's changes

dlqs committed:
Merge pull request #195 from dlqs/bugfix

Bugfix

afterdusk committed:
Remove PE-specific considerations for PPP

Also changed the default HTML font from Noto Serif to Roboto.

afterdusk committed:
Merge pull request #197 from afterdusk/master

Remove PE-specific considerations for PPP
